distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso

```diff
@@ -133,19 +133,39 @@ type Json
              example_unwrap = Json.Number 3 . unwrap
     unwrap : Any
     unwrap self = case self of
-        Json.Array its -> its.map .unwrap
-        Json.Boolean b -> b
-        Json.Number n -> n
-        Json.String t -> t
-        Json.Null -> Nothing
-        Json.Object f -> f.map .unwrap
+        Array its -> its.map .unwrap
+        Boolean b -> b
+        Number n -> n
+        String t -> t
+        Null -> Nothing
+        Object f -> f.map .unwrap
+
+    ## Gets the value associated with the given key in this object.
+
+       Arguments:
+       - field: The name of the field from which to get the value.
+
+       Throws `Nothing` if the associated key is not defined.
+
+       > Example
+         Get the "title" field from this JSON representing a book.
+
+             import Standard.Base.Data.Json
+             import Standard.Examples
+
+             example_get = Examples.json_object.get "title"
+    get : Text -> Json ! No_Such_Field_Error
+    get self field = self.fields.get field . map_error case _ of
+        Map.No_Value_For_Key_Error_Data _ -> No_Such_Field_Error_Data field
+        x -> x
```

Comment:
This function used to be defined (if I understand correctly) only for the `Object` atom.
Now (IIUC) it is defined for all `Json` atoms - but it will still only work for `Object` as others don't have the `fields` field.
I'd suggest adding a pattern match
```
case self of 
    Object -> ... 
    _ -> Error.throw (Illegal_Argument_Error "or something")
```
so that we get a slightly clearer error message when the method is called on other variants.
Before, IIUC, we'd get a method-not-defined error when calling `get`. Now we will get `fields` not defined error which will not really explain to the user what went wrong, so I think it would be good to ensure that the error describes that `get` only works with `Objects`, much like the old error used to.

Reply:
Fair